src/Internal/ObjectExporter/ClosureExporter.php

```diff
@@ -145,9 +164,52 @@ private function getClosure(array $ast, string $file, int $line, array $path) :
         $closure = $closures[0];
 
         if ($closure->uses) {
-            throw new ExportException("The closure has bound variables through 'use', this is not supported.", $path);
+            $this->closureHandleUses($reflectionFunction, $closure, $path);
         }
 
         return $closure;
     }
+
+    /**
+     * Handle `use` part of closure.
+     *
+     * @param ReflectionFunction $reflectionFunction  Reflection of the closure.
+     * @param Node\Expr\Closure  $closure             Parsed closure.
+     * @param string[]           $path                The path to the closure in the array/object graph.
+     *
+     * @throws ExportException
+     */
+    private function closureHandleUses(
+        ReflectionFunction $reflectionFunction,
+        Node\Expr\Closure $closure,
+        array $path
+    ) : void {
+        if (! $this->exporter->closureUseAsVars) {
+            throw new ExportException("The closure has bound variables through 'use', this is not supported.", $path);
```

Comment:
Could you please update the message to say that it's not supported *by default*, but that you can use the `CLOSURE_SNAPSHOT_USE` option to export them?